pkg/collector/container.go

```diff
@@ -121,7 +121,7 @@ func Container(cfg config.Config, logger logr.Logger, otelcol v1alpha1.OpenTelem
 
 	var livenessProbe *corev1.Probe
 	if config, err := adapters.ConfigFromString(otelcol.Spec.Config); err == nil {
-		if probe, err := adapters.ConfigToContainerProbe(config); err == nil {
+		if probe, err := getLivenessProbe(config, otelcol.Spec.LivenessProbe); err == nil {
```

Comment:
Could be a followup PR since it is today's functionality, but should we log the error returned?

Reply:
Since I added validator, I also put a log here.